iModelCore/ECDb/ECDb/ECSql/ECSqlParser.cpp

```diff
@@ -4163,13 +4345,39 @@ ECSqlParseContext::ParseArg const* ECSqlParseContext::CurrentArg() const
 //+---------------+---------------+---------------+---------------+---------------+--------
 void ECSqlParseContext::PopArg() { m_finalizeParseArgs.pop_back(); }
 
+//Helper for TryResolveClass to check if te current node is a table node inside a type predicate
+//If the class is a custom attribute class,we allow it to be used despite what the policy says
+//Example syntax in ECSQL: "IS (mySchema.MyCustomAttribute)"
+bool IsTableNameInTypePredicateSpecialCase(ClassMap const& classMap, OSQLParseNode const& node)

```

Comment:
@khanaffan I recall you had a comment about this section, but we never got to talk about it. Do you have a better way in mind?